src/libs/actions/IOU.js

```diff
@@ -621,12 +621,16 @@ function getMoneyRequestInformation(
  * @param {Boolean} [billable]
  */
 function createDistanceRequest(report, participant, comment, created, transactionID, category, tag, amount, currency, merchant, billable) {
+    // If the report is an iou or expense report, we should get the linked chat report to be passed to the getMoneyRequestInformation function
+    const isMoneyRequestReport = ReportUtils.isMoneyRequestReport(report);
+    const currentChatReport = isMoneyRequestReport ? ReportUtils.getReport(report.chatReportID) : report;
```

Comment:
I think this is an incorrect change. Why not pass the proper report object to `createDistanceRequest()` in the first place?

Reply:
We use the same approach when creating a regular money request:
https://github.com/Expensify/App/blob/1dc45939d71a3888b7091cd1aad8e94563996021/src/libs/actions/IOU.js#L847-L851

What exactly do you mean by a "proper report object" here: is it the one we are fetching as `currentChatReport`? If so, we'll still need to tell apart a request created from a chat and the one created from an expense report so that we can navigate the user properly. Would you have any suggestions for it?

Reply:
Just because we do it somewhere else, doesn't mean it's right. It just means that you get to fix it in two places 🎉 

What I am saying is that the logic that you added inside the method should have been done outside of the method, and then the proper "chat report" is what is passed to the method in the `report` param.

If you need something else for the navigation, then I would consider:
- Moving navigation out of this method (I also dislike this method is doing navigation, it is an anti-pattern of separating concerns)
- Using a different param to control the navigation specifically

Reply:
Makes sense. Since this PR is already merged, would you want me to create a cleanup one?